src/components/DateTimePicker/index.jsx

```diff
@@ -26,32 +26,72 @@ const DateTimePicker = ({
   value,
   labelProps,
   required = false,
-  ...otherProps
+  id,
+  onDatePickerBlur = noop,
+  onTimePickerBlur = noop,
+  onDatePickerFocus = noop,
+  onTimePickerFocus = noop,
+  datePickerProps,
+  timePickerProps,
+  autoUpdateTime = true,
 }) => {
   const [open, setOpen] = React.useState(false);
   const [date, setDate] = useState();
   const [time, setTime] = useState();
+  const [isHourFocused, setIsHourFocused] = useState(false);
+  const [isMinuteFocused, setIsMinuteFocused] = useState(false);
 
   const timeRef = React.useRef(null);
-  const id = useId(otherProps.id);
-  const errorId = `error_${id}`;
+  const defaultId = useId(id);
+  const errorId = `error_${defaultId}`;
   const handleDateChange = date => {
     setDate(date);
-    setTime(date);
+    if (autoUpdateTime) setTime(date);
```

Comment:
Instead of adding an additional prop, we can check if the value of `time` state here. If its `undefined` or `null` set the state, otherwise do not set the `time` state value.

Also while calling the `onChange` callback make sure we are setting the time value of the object to match the value in the `time` state.